uiowa/uiowa
# FontAwesome Role Presentation

Resolves https://github.com/uiowa/uiowa/issues/1320

Short of hearing back on https://github.com/uiowa/uiowa/issues/1320#issuecomment-846075644, I decided to move forward with the quick wins.

## ToDo

- [x] Event icon field
- [x] Happy config import for both install/sync

## To Test

- Anywhere you see a font awesome icon (except existing footer contact content) you should see the svg markup with `role="presentation"` instead of `role="img"`
  - Places to look:
    - Button arrows
    - Person Card Details
    - Event information
    - Social media menu items
    - ...
- In the WYSIWYG if you place an icon manually `<span class="fa-user fas"></span>` on render it should include `role="presentation"`. If you place an icon manually and try to add a role yourself that isn't presentation, it will override with the correct "presentation" value.
- Search the docroot for icons that I might have missed (our code base, not contrib).
- **Note:** FontAwesome itself has not been updated as part of this PR so icon classes haven't changed, etc.
- Not sure if any a11y checker (siteimprove, webaim, lighthouse) is still flagging this, but you shouldn't see alt text errors for icons.
